nucliadb/src/nucliadb/common/datamanagers/vectorsets.py

```diff
@@ -68,7 +68,7 @@ async def set(txn: Transaction, *, kbid: str, config: knowledgebox_pb2.VectorSet
 
 
 async def delete(txn: Transaction, *, kbid: str, vectorset_id: str):
-    kb_vectorsets = await _get_or_default(txn, kbid=kbid)
+    kb_vectorsets = await _get_or_default(txn, kbid=kbid, for_update=False)
```

Comment:
```suggestion
    kb_vectorsets = await _get_or_default(txn, kbid=kbid, for_update=True)
```

This modifies the vectorset list key so it needs for_update even if it's a deletion. I think we talked about this and I gave you the wrong answer, sorry.